ZelBack/src/services/utils/pathSecurity.js

```diff
@@ -0,0 +1,354 @@
+    if (component === '' && i === components.length - 1) {
+      continue;
+    }
+
+    // Empty component in middle means double slash - suspicious
+    if (component === '' && i < components.length - 1) {
+      throw new Error('Invalid path: empty path component (double slash)');
+    }
+
+    if (!isValidPathComponent(component)) {
+      throw new Error(`Invalid path: component "${component}" contains disallowed characters`);
+    }
+  }
+}
+
+/**
+ * Check if a path component is a traversal attempt.
+ * This is used in non-strict mode for basic traversal detection.
+ *
+ * @param {string} component - Path component to check
+ * @returns {boolean} True if component is a traversal attempt
+ */
+function isTraversalComponent(component) {
+  return component === '.' || component === '..';
+}
+
+/**
+ * Basic path validation (non-strict mode).
+ * Only blocks obvious dangerous patterns without character restrictions.
+ *
+ * @param {string} userPath - User-provided path
+ * @throws {Error} If path contains traversal attempts
+ */
+function validatePathBasic(userPath) {
+  if (!userPath || typeof userPath !== 'string') {
+    return;
+  }
+
+  const normalizedPath = normalizePathSeparators(userPath);
+  const components = normalizedPath.split('/');
+
+  for (const component of components) {
+    if (isTraversalComponent(component)) {
+      throw new Error('Invalid path: directory traversal not allowed');
+    }
+  }
+}
+
+/**
+ * Sanitize and validate a relative path to prevent directory traversal.
+ * Uses defense-in-depth with both blocklist and allowlist validation.
+ * Ensures the resolved path stays within the allowed base directory.
+ *
+ * @param {string} userPath - User-provided relative path
+ * @param {string} basePath - The allowed base directory
+ * @param {object} options - Optional configuration
+ * @param {boolean} options.strict - If true, enforce strict allowlist (default: true)
+ * @returns {string} Resolved safe absolute path
+ * @throws {Error} If path attempts to escape base directory or contains invalid characters
+ *
+ * @example
+ * // Valid usage
+ * sanitizePath('appdata', '/apps/myapp')  // Returns '/apps/myapp/appdata'
+ * sanitizePath('logs/app.log', '/apps/myapp')  // Returns '/apps/myapp/logs/app.log'
+ * sanitizePath('', '/apps/myapp')  // Returns '/apps/myapp'
+ *
+ * // Throws error
+ * sanitizePath('..', '/apps/myapp')  // Error: directory traversal
+ * sanitizePath('../other', '/apps/myapp')  // Error: directory traversal
+ * sanitizePath('foo\0bar', '/apps/myapp')  // Error: null byte
+ */
+function sanitizePath(userPath, basePath, options = {}) {
+  const { strict = true } = options;
+
+  // If no user path provided, return the base path
+  if (!userPath || typeof userPath !== 'string') {
+    return basePath;
+  }
+
+  // === LAYER 1: Blocklist checks (fast rejection of known-bad patterns) ===
+
+  // Check for null bytes (can bypass security checks in some systems)
+  if (userPath.includes('\0')) {
```

Comment:
This should just check for backslashes, then raise (and we delete normalizePathSeparators)